// @grammyjs/storage-prisma/dist/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.PrismaAdapter = void 0;
__exportStar(require("./types/SessionDelegate"), exports);
class PrismaAdapter {
    constructor(repository) {
        this.sessionDelegate = repository;
    }
    async read(key) {
        const session = await this.sessionDelegate.findUnique({ where: { key } });
        return (session === null || session === void 0 ? void 0 : session.value) ? JSON.parse(session.value) : undefined;
    }
    async write(key, data) {
        const value = JSON.stringify(data);
        await this.sessionDelegate.upsert({
            where: { key },
            create: { key, value },
            update: { value },
        });
    }
    async delete(key) {
        await this.sessionDelegate.delete({ where: { key } });
    }
}
exports.PrismaAdapter = PrismaAdapter;
